perceval/backends/remo2.py

```diff
@@ -0,0 +1,318 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import json
+import logging
+import os.path
+
+import requests
+
+from dateutil import parser
+
+from ..backend import Backend, BackendCommand, metadata
+from ..cache import Cache
+from ..errors import BackendError, CacheError, ParseError
+
+from ..utils import (DEFAULT_DATETIME,
+                     datetime_to_utc,
+                     str_to_datetime,
+                     urljoin)
+
+
+logger = logging.getLogger(__name__)
+
+MOZILLA_REPS_URL = "https://reps.mozilla.org"
+
+class ReMo(Backend):
+    """ReMo backend for Perceval.
+
+    This class retrieves the events from a
+    ReMo url. To initialize this class a
+    url could be provided. If not, https://reps.mozilla.org will be used.
+
+    It uses v2 API to get events, people and activities data.
+
+    :param url: ReMo url
+    :param cache: cache object to store raw data
+    :param origin: identifier of the repository; when `None` or an
+        empty string are given, it will be set to `url` value
+    """
+    version = '0.2.0'
+
+    def __init__(self, url=None, cache=None, origin=None):
+        if not url:
+            url = MOZILLA_REPS_URL
+        origin = origin if origin else url
+
+        super().__init__(origin, cache=cache)
+        self.url = url
+        self.client = ReMoClient(url)
+        self.__users = {}  # internal users cache
+
+    @metadata
+    def fetch(self, kind='events'):
+        """Fetch events from the ReMo url.
+
+        The method retrieves, from a ReMo url, the
+        events.
+
+        :returns: a generator of events
+        """
+
+        supported_kinds = ['activities', 'events', 'users']
+
+        if kind not in supported_kinds:
+            raise RuntimeError('ReMo perceval backend does not support %s', kind)
```

Comment:
This statement should raise a `ValueError` exception instead of a `RuntimeError`. Moreover, I think you forgot to concatenate `kind` with the previous string.